### PR TITLE
[DOCS] Document `throttle_period_in_millis` for watcher actions

### LINKED ISSUE #45346: [DOCS] Add examples in watchers definitions that uses throttle_period_in_millis
**Description of the problem including expected versus actual behavior**:

When defining an action in a watcher you can configure a *throttle period*. 

In the documentation all the examples use `throttle_period` parameter, but you could use `throttle_period_in_millis` instead, since, as per the [code](https://github.com/elastic/elasticsearch/blob/v7.3.0/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/watch/WatchParser.java#L158-L167) both parameters are accepted. 
However, `throttle_period_in_millis` parameter is mentioned neither in the examples related to watchers in the Elasticsearch documentation nor anywhere in the docs. 

*Examples in the doc using **throttle_period***
https://www.elastic.co/guide/en/elastic-stack-overview/7.2/actions-slack.html
https://www.elastic.co/guide/en/elastic-stack-overview/7.1/actions.html#actions-ack-throttle

*Example in the discussion forum using **throttle_period_in_millis***
https://discuss.elastic.co/t/how-to-set-a-time-range-in-the-watcher/93679/4

 It would be good to include examples in the documentation using  **throttle_period_in_millis** too.

### PR DESCRIPTION
**Pull request by jrodewig**
Closes #45346.

### Preview
http://elasticsearch_45607.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/watcher-api-put-watch.html#_request_body_7

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by szabosteve — APPROVED** [review]
LGTM.

File: x-pack/docs/en/rest-api/watcher/put-watch.asciidoc [hunk 1]
```diff
@@ -68,7 +68,15 @@ A watch has the following fields:
 
 | `throttle_period` | The minimum time between actions being run, the default
                       for this is 5 seconds. This default can be changed in the
-                      config file with the setting `xpack.watcher.throttle.period.default_period`.
+                      config file with the setting
+                      `xpack.watcher.throttle.period.default_period`. If
```
Comment (hub-cap) [inline x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: Hey i did a quick eyeball of the code here and it looks like this is not correct. The code is basically "last one wins" due to the for loop + non final variable assignment when parsing the watch.

So if you have 
```
"throttle_period_in_millis": "7000",
"throttle_period": "5s",
```
the GET watch API returns `throttle_period_in_millis":5000,`

and if you have the opposite, 
```
"throttle_period": "5s",
"throttle_period_in_millis": "7000",
```
the GET watch API returns `throttle_period_in_millis":7000,`

File: x-pack/docs/en/rest-api/watcher/put-watch.asciidoc [hunk 2]
```diff
@@ -68,7 +68,15 @@ A watch has the following fields:
...
+                      provided, the `throttle_period_in_millis` value overrides
+                      this parameter.
+
+| `throttle_period_in_millis` | Minimum time in milliseconds between actions
+                                being run. Defaults to `5000`. If provided, this
+                                value overrides the `throttle_period` value.
```
Comment (hub-cap) [inline x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: same here WRT the overrides.

**Review by hub-cap — CHANGES_REQUESTED** [review]
other than changing the wording on the overrides, this is LGTM.

File: x-pack/docs/en/rest-api/watcher/put-watch.asciidoc (hunk 1 above)
Comment (hub-cap) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: Here is the code im talking about for reference, 

https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/watch/WatchParser.java#L159-L163

Comment (jrodewig) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: Tricky! I did look at that code, but I incorrectly assumed the first matching `else if` would win.

I reversed the override statements with cfe41d9cb17ef082fe8f314779ad5f9402bafa93.

Thanks for catching this @hub-cap.

File: x-pack/docs/en/rest-api/watcher/put-watch.asciidoc (hunk 2 above)
Comment (jrodewig) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: Fixed with cfe41d9.

File: x-pack/docs/en/rest-api/watcher/put-watch.asciidoc (hunk 1 above)
Comment (hub-cap) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: So its not about one overriding the other, its about which one is declared first in the watch. if you declare both, the last one overrides the first one, so you can see in the example above that when i declared them both, which i did in 2 watches, the output was different based on which order they were declared in.

Comment (jrodewig) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: 🤦‍♂  Sorry about that. Guess I'm feeling dense today.

Corrected again with deaef45ab4202c50b15d6acac4abfaa036ecb119.

File: x-pack/docs/en/rest-api/watcher/put-watch.asciidoc (hunk 2 above)
Comment (jrodewig) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: Corrected again with deaef45.

Comment (hub-cap) [reply x-pack/docs/en/rest-api/watcher/put-watch.asciidoc]: :boom:

**Review by hub-cap — APPROVED** [review]

**Pull request merged**